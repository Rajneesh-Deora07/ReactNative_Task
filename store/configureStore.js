import {createStore, applyMiddleware} from 'redux';
import rootReducer from './Index';
import createSagaMiddleware from 'redux-saga';
import rootSaga from '../Sagas/Sagas';

export const configureStore = (initialState) => {
  const sagaMiddleware = createSagaMiddleware();
  const middlewares = [sagaMiddleware];

  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(...middlewares),
  );

  sagaMiddleware.run(rootSaga);

  return store;
};
