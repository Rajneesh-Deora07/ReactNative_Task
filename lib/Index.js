import React from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {
  StyleSheet,
  View,
  ActivityIndicator,
  Dimensions,
  Alert,
} from 'react-native';

import WeatherInput from './WeatherInput';
import WeatherList from './WeatherList';
import {startFetchData} from '../store/action';

const ForecastWeather = () => {
  const weather = useSelector((state) => state.weather);
  const forecastData = weather.data || {};
  const isDataFetching = weather.isFetching;
  const dispatch = useDispatch();

  const callWeatherAPI = (zipCode) => {
    if (zipCode.length > 0) {
      dispatch(startFetchData(zipCode));
    } else {
      Alert.alert('Please enter valid ZipCode');
    }
  };

  return (
    <View style={styles.containerStyle}>
      <WeatherInput fetchData={callWeatherAPI} />
      <WeatherList data={forecastData} />
      {!isDataFetching ? null : (
        <View style={styles.container}>
          <ActivityIndicator size="large" color="#grey" />
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    flex: 1,
    alignItems: 'center',
  },
  container: {
    top: '50%',
    bottom: '50%',
    position: 'absolute',
  },
  horizontal: {
    backgroundColor: 'transparent',
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
  },
});
export default ForecastWeather;
export {ForecastWeather as ForecastWeatherObject};
