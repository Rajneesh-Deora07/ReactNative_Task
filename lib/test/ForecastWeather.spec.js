import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import {ForecastWeatherObject} from '../Index';
import * as redux from 'react-redux';

import {View} from 'react-native';
import {
  useSelector as originalUseSelector,
  useDispatch as originalUseDispatch,
} from 'react-redux';

export const useSelector = (state) => originalUseSelector(state);
export const useDispatch = () => originalUseDispatch();
describe('Forecast test cases ------', () => {
  let wrapper;
  let store;
  let data = [
    {
      id: 15948948,
      dt_text: '2021-03-09 15:00:00',
      wind: {speed: 1.71},
      weather: [{description: 'clear sky'}],
      main: {humidity: 14, temp: 302.07},
    },
  ];
  beforeEach(() => {
    store = configureStore()({
      data: data[0],
      isLoading: false,
      error: null,
    });
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((state) => store.getState());

    jest.spyOn(redux, 'useDispatch').mockImplementation(() => store.dispatch);

    wrapper = shallow(<ForecastWeatherObject store={store} />);
  });

  it('Snapshot match', () => {
    console.log('wrapper', wrapper);
  });

  it('Render UI of ForecastWeather', () => {
    expect(wrapper.find(View).length).toBe(1);
    expect(wrapper.find('WeatherInput').length).toBe(1);
    expect(wrapper.find('WeatherList').length).toBe(1);
  });
});
