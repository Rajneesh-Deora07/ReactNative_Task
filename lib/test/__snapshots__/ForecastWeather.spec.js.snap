// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast test cases ------ Snapshot match 1`] = `ShallowWrapper {}`;
